src/ExternalImage.php

```diff
@@ -0,0 +1,479 @@
+ *   <div class="body">
+ *     Whatever whatever
+ *   </div>
+ *   <img
+ *     src="http://example.org/wp-content/uploads/2015/06/kurt.jpg"
+ *     alt="Another way to initialize images as Timber\Image objects, but within Twig" />
+ * </article>
+ * ```
+ */
+class ExternalImage implements ImageInterface
+{
+    /**
+     * File.
+     *
+     * @api
+     * @var string
+     */
+    private $alt_text;
+
+    protected function __construct($alt)
+    {
+        $this->alt_text = $alt;
+    }
+
+    /**
+     * File.
+     *
+     * @api
+     * @var mixed
+     */
+    public $file;
+
+    /**
+     * File location.
+     *
+     * @api
+     * @var string The absolute path to the attachmend file in the filesystem
+     *             (Example: `/var/www/htdocs/wp-content/uploads/2015/08/my-pic.jpg`)
+     */
+    public $file_loc;
+
+    /**
+     * Absolute URL.
+     *
+     * @var string The absolute URL to the attachment.
+     */
+    public $abs_url;
+
+    /**
+     * File extension.
+     *
+     * @api
+     * @since 2.0.0
+     * @var null|string A file extension.
+     */
+    public $file_extension = null;
+
+    /**
+     * Formatted file size.
+     *
+     * @api
+     * @since 2.0.0
+     * @var FileSize File size string.
+     */
+    private $file_size = null;
+
+    /**
+     * File types.
+     *
+     * @var array An array of supported relative file types.
+     */
+    private $image_file_types = [
+        'jpg',
+        'jpeg',
+        'png',
+        'svg',
+        'bmp',
+        'ico',
+        'gif',
+        'tiff',
+        'pdf',
+    ];
+
+    /**
+     * Image dimensions.
+     *
+     * @internal
+     * @var ImageDimensions stores Image Dimensions in a structured way.
+     */
+    protected ImageDimensions $imageDimensions;
+
+    /**
+     * Inits the ExternalImage object.
+     *
+     * @param $url string URL to load the image from.
+     * @param $alt string ALT text for the image.
+     * @internal
+     *
+     */
+    public static function build($url, $alt)
+    {
+        if (!is_numeric($url) && is_string($url)) {
+            $external_image = new ExternalImage($alt);
+            if (strstr($url, '://')) {
+                // Assume URL.
+                $external_image->init_with_url($url);
+
+                return $external_image;
+            } elseif (strstr($url, ABSPATH)) {
+                // Assume absolute path.
+                $external_image->init_with_file_path($url);
+
+                return $external_image;
+            } else {
+                // Check for image file types.
+                foreach ($external_image->image_file_types as $type) {
+                    // Assume a relative path.
+                    if (strstr(strtolower($url), $type)) {
+                        $external_image->init_with_relative_path($url);
+
+                        return $external_image;
+                    }
+                }
+            }
+        }
+
+        return null;
```

Comment:
We should probably also return `null` if the local file doesn’t exist.

I guess we don’t have any check whether the file exists yet in the whole logic. Would it be good to have that?

Reply:
Both for URLs and local files?

Reply:
@xavivars I wouldn’t check external URLs, but since external URLs will be saved in the uploads folder and become local files when sideloaded, we could probably check those.

Reply:
Thinking further about this, I guess it’s not so bad if an image is displayed as a broken image on the website. That’s also what alt texts are good for. If we just return `null`, then it would be hard to track these images.

Let’s leave it like that and see if we can find a good solution in the future.